engine/access/rpc/backend/backend_execution_data.go

```diff
@@ -0,0 +1,41 @@
+package backend
+
+import (
+	"context"
+
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/state_synchronization/requester/jobs"
+	"github.com/onflow/flow-go/storage"
+)
+
+type backendExecutionData struct {
+	blocks   storage.Blocks
+	execData jobs.ExecutionDataReader
```

Comment:
I don't think you should use `jobs.ExecutionDataReader` directly as that's build specifically for use within the jobqueue framework, and ends up doing a few extra database queries than necessary. Instead, you just need the few lines of code from [`getExecutionData` method](https://github.com/onflow/flow-go/blob/master/module/state_synchronization/requester/jobs/execution_data_reader.go#L108-L127)